casper/src/main/resources/PoS.rhox

```diff
@@ -77,6 +96,21 @@ new  PoS,
               }
             } |
 
+            contract PoS (@"getActiveValidators", returnCh) = {
+              new tmpCh in {
+                getMVar!(*stateCh, *tmpCh) |
+                for (@state <- tmpCh) {
+                  returnCh!(state.get("activeValidators"))
+                }
+              }
+            } |
+
+            contract PoS (@"getActiveValidatorVaults", returnCh) = {
```

Comment:
I don't think this should be part of the PoS API. The fact that we have a vault per validator should be only an implementation detail

Reply:
I agree, but I don't see  another way to access the vaults in Rholang. One hack would be to interpolate unforgeable names into the Rholang source code (for the necessary methods) and call those names through system processes in the tests. We might have to do that with some of the current public methods in `PoS` that will need to be private in production. If I'm not mistaken, `closeBlock` is an example.

Reply:
We would extract the "vault per validator" logic into a separate vault-like contract that would have an address and a `balance` method, the same as all the "normal" vaults. The PoS contract will not really care about the actual implementation of its vault and the vault-per-validator logic can be tested outside of PoS tests.